packages/desktop-client/src/components/reports/reports/Spending.tsx

```diff
@@ -224,13 +229,23 @@ export function Spending() {
                 >
                   Compare this month to:
                 </Text>
-                <ModeButton
-                  selected={mode === 'Last month'}
-                  onSelect={() => setMode('Last month')}
-                >
-                  Last month
-                </ModeButton>
-                {showAverage && (
+                {showLastMonth && (
```

Comment:
Rather than have `showLastMonth` checks in every element, it might be better to do just one at the top level `<View>` (line 219).

```
{!showLastMonth ? (
  <View>
    message about need for more data
  </View>
) : (
  <>
    Existing code for buttons and graph display
  </>
)}
```

Reply:
This is still an issue. Here's an example:

```
              {!showLastMonth ? (
                <View style={{ marginTop: 30 }}>
                  <h1>Additional data required to generate graph</h1>
                  <Paragraph>
                    Currently, there is insufficient data to display any
                    information regarding your spending. Please input
                    transactions from last month to enable graph visualization.
                  </Paragraph>
                </View>
              ) : (
                <>
                  <View
                    style={{
                      alignItems: 'center',
                      flexDirection: 'row',
                    }}
                  >
                    <Text
                      style={{
                        paddingRight: 10,
                      }}
                    >
                      Compare this month to:
                    </Text>
                    <ModeButton
                      selected={mode === 'Last month'}
                      onSelect={() => setMode('Last month')}
                    >
                      Last month
                    </ModeButton>
                    {showLastYear && (
                      <ModeButton
                        selected={mode === 'Last year'}
                        onSelect={() => setMode('Last year')}
                      >
                        Last year
                      </ModeButton>
                    )}
                    {showAverage && (
                      <ModeButton
                        selected={mode === 'Average'}
                        onSelect={() => setMode('Average')}
                      >
                        Average
                      </ModeButton>
                    )}
                  </View>

                  {dataCheck ? (
                    <SpendingGraph
                      style={{ flexGrow: 1 }}
                      compact={false}
                      data={data}
                      mode={mode}
                    />
                  ) : (
                    <LoadingIndicator message="Loading report..." />
                  )}
                </>
              )}
```